### DISCUSSION TITLE
How do I transfer contributions to another account

### DISCUSSION THREAD

**Discussion by catalincatrina** [community/community#23075] [New to GitHub]
<p>I am talking about the contributions that show up as a grid in my user profile. I have two accounts, one for work, one for personal use, and I want for my work account contributions to show on my personal account as well.</p>
<p>The reason for that is because once I will not work for the current company anymore and they will close/delete my email, I will lose access to that account, I will be left with only my personal account, and it will appear as if I didn’t do any work, for future employers.</p>
<p>How do I link the activity from one account to another, both being mine? One of them has as primary email my personal email, and the other has as primary email the email I was given from work.</p>

**Comment by jfullstackdev** [comment]
<p>just simply add the email tied to those commits in the other account,</p>
<p>of course, it should be deleted first at the user that you want to abandon</p>
<p>I’m not very sure when the company deleted your work email or whether you can still claim it, maybe you can ask here</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://support.github.com/contact/account">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/47092464/181074708-597d44b9-e5b6-489c-b86c-4cfc2ecff2ae.png" class="site-icon" width="32" height="32">

      <a href="https://support.github.com/contact/account" target="_blank" rel="noopener">support.github.com</a>
  </header>

  <article class="onebox-body">
    

<h3><a href="https://support.github.com/contact/account" target="_blank" rel="noopener">Accounts - GitHub Support</a></h3>

  <p>GitHub Support is here to help. Learn about GitHub products, browse our helpful resources, and contact support with your questions.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by catalincatrina** [comment]
<p>My company didn’t delete my email yet, but they will most likely do it once I stop working there in the future.</p>
<p>So I need to first delete the work email from the work account, and add the work email into my personal account. I guess I will do that at the very end before resigning.</p>
<p>What about when I’ll lose access to the work email though? What happens when they will eventually delete it? Will the contributions dissapear as well?</p>

**Comment by jfullstackdev** [comment]
<aside class="quote no-group" data-username="catalincatrina" data-post="3" data-topic="257835">
<div class="title">
<div class="quote-controls"></div>
 catalincatrina:</div>
<blockquote>
<p>What about when I’ll lose access to the work email though?</p>
</blockquote>
</aside>
<p>when you just lose the access, you can still add it unverified and the commits will be added</p>
<aside class="quote no-group" data-username="catalincatrina" data-post="3" data-topic="257835">
<div class="title">
<div class="quote-controls"></div>
 catalincatrina:</div>
<blockquote>
<p>What happens when they will eventually delete it?</p>
</blockquote>
</aside>
<p>this is I’m not very sure of that’s why I said try asking it at contact</p>
<p><a href="https://support.github.com/contact/account">https://support.github.com/contact/account</a></p>
<p>or maybe some member of the GitHub Staff here knows the answer and will comment,</p>

**Reply by skverma618** [reply]
So is it really possible to transfer git history from one account to another?

**Comment by queenofcorgis** [comment]
<p>:wave:t3: <a class="mention" href="/catalincatrina">@catalincatrina</a> - was <a class="mention" href="/jdevstatic">@jdevstatic</a> able to assist here, if so don’t forget to mark their comments as the solution for any other users who may encounter this, thanks!</p>

**Comment by catalincatrina** [comment]
<p>He definitely helped but there is still the edge case about what happens if my company will delete my email after I leave? will my commits / history/ contributions on the github account associated with that email, stay?</p>

**Comment by queenofcorgis** [comment, marked as answer]
<p>Thanks for asking:  It doesn’t need to be a working email address, or to be verified, in order to match commits to your profile as contributions so they should stay.</p>

**Reply by amddevmh** [reply]
why so many downvotes here?

**Reply by sglover-onemedical** [reply]
it seems like maybe it doesn't work?

**Reply by siddhantsingh-25** [reply]
Now, i have no clue, whether to move forward with it or not. Someone pls help!!

**Comment by irfanbabar** [comment]
@jdevstatic  I m stuck, how to remove the email address of the company from the company GitHub without adding another email address?

what I understand from above is that in order to add my work email address to my personal Github account I have to remove it first from my work account.

**Reply by CriticalGnome** [reply]
> @jdevstatic I m stuck, how to remove the email address of the company from the company GitHub without adding another email address?
> 
> what I understand from above is that in order to add my work email address to my personal Github account I have to remove it first from my work account.

You still can delete old GitHub account

**Comment by karkinissan** [comment]
Hi, I want to do this as well. How was this resolved?

**Comment by Aideepakchaudhary** [comment]
@catalincatrina how do you solve this ?

**Comment by AimoreRRD** [comment]
I also wasn't able to get my activity by removing the email from my business account and adding it to my personal email.
Do I have to wait for them to appear? Or does it mean that old activity with that account cannot be transferred, only future activity?

**Reply by samyosm** [reply]
Any updates? Were you able to?

**Comment by RahulSomaliya** [comment]
@queenofcorgis, I've got the same issue as well please provide more clarity, my concerns are;

- My company's repo is a private one like most of us, so will I be able to get contributions anonymously without risking their codebase? 
- If yes, how do I resolve `this email already in use` issue while adding the company email to my personal account? some guys here suggested deleting the company email account from GitHub first, is that the correct way of doing it? won't that action lead to deleting my contributions altogether? If not, or there is some other way - please share me a precise guide for doing it. 

Anyone who's already done this please help, this is important!

**Reply by andrescn20** [reply]
Hello. I did just what you said: I removed the company email from my work account. I then add that same email to my personal account. And it automatically added all my previous work commits to my personal account. The only problem is, there was a time when I had the email privacy checked, and seems like all commits associated with the "no reply" email given by GitHub were not reassigned. But all commits originally associated with my work account (directly) are now properly associated with my personal account.

**Reply by sumanabhi** [reply]
@andrescn20 Is the verification required from Work Email? or just adding the email id is fine for the contribution association to personal account.

How much time does it take if it doesn't required the Verification from the Work account.

**Reply by andrescn20** [reply]
Hello @sumanabhi. I don't actually remember that particular detail from my own process. I do remember reading in this thread and others that verification is not necessary, as long as the email is freed from any other account and added to your current account.

**Reply by sumanabhi** [reply]
@andrescn20 I have done it but it is not getting reflecting in my account.

**Reply by andrescn20** [reply]
Have you checked that the contributions are associated with that email? I had a problem with several of my contributions because I had enabled the option to hide my email. Because of this, my contributions were not associated with my email but with a new "no-reply" email made by GitHub. It would be great for you to verify this.

**Reply by sumanabhi** [reply]
Okay Let me check that But I'm pretty sure that it was associated with the email only. let me check once again.

**Reply by AnujCMD** [reply]
@sumanabhi My contributions are also not reflecting up, did you found any solution?

**Reply by sumanabhi** [reply]
@AnujCMD Not yet. I didn't find this solution working as well. because as soon you add some another email there in that work account then automatically all the commits got linked with that one. and that's the main reason that is not reflected in our current account. 

I guess it will only work if the work github account got deleted.

**Reply by AashishKumar-3002** [reply]
Hey , what was the final solution. when I try adding my email into my account it says already in use and when i try removing my email from the work git account it says needs at least one email. Will i have access to the private repos of my organization

**Reply by sumanabhi** [reply]
@All I guess it is gonna work only in the case when your work account must be deleted. 

Even I tried all the above steps but nothing is working. I'm still open to get some concrete solutions around this.

**Comment by doomroep5** [comment]
> I am talking about the contributions that show up as a grid in my user profile. I have two accounts, one for work, one for personal use, and I want for my work account contributions to show on my personal account as well.
> 
> The reason for that is because once I will not work for the current company anymore and they will close/delete my email, I will lose access to that account, I will be left with only my personal account, and it will appear as if I didn’t do any work, for future employers.
> 
> How do I link the activity from one account to another, both being mine? One of them has as primary email my personal email, and the other has as primary email the email I was given from work.

**Reply by andrescn20** [reply]
The only solution is to add the work email (already associated with your commits) to your personal account. In order to achieve this you have to "free" your work email. You can do this simply by adding a new primary email to your work account and then deleting the one you were previously using. Now that this email is free, you can add it to your personal account. Oh and yes, the commits are the contributions. As long as your contributions are associated with the email you just moved, they will be updated on your personal account. Btw, I did it exactly for the same reason you are planning on doing it.

**Reply by AashishKumar-3002** [reply]
Hey , what was the final solution. when I try adding my email into my account it says already in use and when i try removing my email from the work git account it says needs at least one email.

**Comment by alvingalit** [comment]
I created a new github account using my work email, too. I was wondering if there is a period after deleting this account which will keep me from adding my work email to my personal account?

**Comment by hugomiranda94** [comment]
How long did it take for your contributions from the "work profile" to be shown on your "personal profile"? @catalincatrina

**Reply by cucupac** [reply]
Any answer to this? I'm in the same boat.

**Reply by hugomiranda94** [reply]
I don't have them, I did set up my work email into my personal profile but no contributions. I don't think it works, maybe it will until I delete my work account, but I'm still working at the same company; my idea was to "merge" the two accounts into one, but no luck unfortunately

**Reply by thomaschow** [reply]
Hi, ping on this question? I've tried adding my work email to my personal github account after assigning my work account a temporary email, but I still don't see my work contributions showing up on my personal profile.

**Reply by hugomiranda94** [reply]
I was finally able to see my contributions after setting the "Include private contributions on my profile" to true. This works as long as the email with which you made contributions is set on your account and verified as well, so you will need to have access to that email to accept GitHub's verification email once you add it to your account.

**Comment by sglover-onemedical** [comment]
Did anyone get this to work (e.g. get contributions to one personal account to accrue on one main personal account)?

**Comment by jdvivar** [comment]
Have you tried with this projects?
https://github.com/kefimochi/sync-contribution-graph
https://github.com/charpeni/sync-external-contributions

**Comment by Mohammad-Almosallam** [comment]
## **So here is the solution✨ (No workarounds)**
I have tried all the above solutions, but none worked for me. It definitely helped me understand the issue very well, I even contacted Github support to get more information on this issue. 

So, If you are a person who has **work account** and wants to transfer commits to **personal account** , this is my solution for you:

### **The idea:**
You need to detach the work email from the work account, and attach it to your personal account. That is how you will be able to get your work contributions transferred to your personal account.

### **How?** 
**Step 1:** 
Github allows you to add multiple emails in one account, This is only under the condition that this email is **free**, meaning it isn't under other account (e.g. your work account). Hence, **in order to transfer our work email to our personal account we need to set it free first**. 

This can be done as follows:
1. Create a new email that isnt linked to any github account so that it acts as a placeholder
2. Add the new email into your work account. _(Settings > Emails > Add email address field)_
3. Set the new email as primary email for your work account. _(Settings > Emails > Primary email address field)_
4. **Free** the **work email** from work account by clicking on the trash icon

Now your work email that has all your contributions is **free!**, so it can be linked to your personal account!

**Step 2**

1. Sign out from your work account.
2. Log in to your personal account
3. Add your work email to your personal account but do not set it to primary.

Congrats🎉, your contributions will now appear in the graph.👏🏻 **If it didnt appear in graph.**. then there is a high probability that **_Keep my email addresses private_** is checked. which means you set web-based Git operations (e.g. edits and merges) to be sent from github. **But don't worry this is simple and easy to fix.** 

Go to _(Settings > Emails)_ , then uncheck  **_Keep my email addresses private_**.

Enjoy❤️
Feel free to ask. 
Resources [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile)

**Reply by emeiese** [reply]
As I understand it, this only works if you are actually leaving your company, right? Because I was searching for a solution that transfers my contributions from my work account to my personal account, however, I don't want to leave my company yet, so I want to keep the work email on the work account. This way, emails related to my work's GitHub reach my work email and not my personal email. 😿

Probably a nice feature for this cases would be to have one master account that gets all the contributions of the sub-accounts, since my company probably don't want to grant access to my personal github account to the company's repos.

**Reply by haydenw2005** [reply]
Question- will it work if my old work address is deactivated/deleted and hence cannot be verified? I followed all the steps and added my old work email to my primary account, but I am not seeing those contributions.

**Reply by icpsoni** [reply]
It did not work for me at first attempt, I relinked the account with previous one disabled "Keep my email addresses private" and did it again.
Works now.

**Reply by junwen-k** [reply]
This is a fantastic and thorough solution—works for me and thank you for sharing! 🙌

I’d like to suggest a neat trick that could make the process even simpler when trying to free up your work email: using [plus addressing](https://learn.microsoft.com/en-us/exchange/recipients-in-exchange-online/plus-addressing-in-exchange-online).

For example, if your work email is `work@org.com`, you can add `work+github@org.com` in your GitHub email settings. Verification emails sent to `work+github@org.com` will still land in your primary inbox at `work@org.com`. This trick helps bypass creating a placeholder email while keeping everything streamlined.

@Mohammad-Almosallam If you think this could be helpful, I’d love to see it added to your steps to benefit more users who might face similar situations. A very neat trick!

**Reply by DaroMacs** [reply]
amazing, thanks!

**Reply by VirajMadhu** [reply]
Well, this solution worked pretty well. It might take some time to sync contributions. Thanks

**Reply by mgk815** [reply]
Hello, thank you for this post! I followed all the following steps and it is still not syncing. How long should it take?

**Reply by m-elagamy** [reply]
I have followed all the steps and still  not syncing and unfortunately i lost most contributes on work account but nothing be added to personal account!

**Before:(old contributes, almost have 950 now)** 

![WhatsApp Image 2025-09-19 at 16 56 13_2e35f0e6](https://github.com/user-attachments/assets/b6a68050-f111-4cd1-8ad3-31aedff4bf02)

Now: 

<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/db3f0b68-ed99-44ac-997d-1c3b3cd3c94f" />

What happened?

**Reply by Rishaaaan** [reply]
> This is a fantastic and thorough solution—works for me and thank you for sharing! 🙌
> 
> I’d like to suggest a neat trick that could make the process even simpler when trying to free up your work email: using [plus addressing](https://learn.microsoft.com/en-us/exchange/recipients-in-exchange-online/plus-addressing-in-exchange-online).
> 
> For example, if your work email is `work@org.com`, you can add `work+github@org.com` in your GitHub email settings. Verification emails sent to `work+github@org.com` will still land in your primary inbox at `work@org.com`. This trick helps bypass creating a placeholder email while keeping everything streamlined.
> 
> @Mohammad-Almosallam If you think this could be helpful, I’d love to see it added to your steps to benefit more users who might face similar situations. A very neat trick!

I followed all these steps but i dont see any contributions added to my profile?

**Comment by jsspen** [comment]
The email trick isn't working for me. I've got my personal email and my work email added on _both_ accounts and it didn't even complain about the email not being "free." I even tried setting them both to "primary" - no luck.

**Comment by hugomiranda94** [comment]
I was able to see my contributions after setting the "Include private contributions on my profile" to true. This works as long as the email with which you made contributions is set on your desired account and verified as well, so you will need to have access to that email to accept GitHub's verification email once you add it to your account.

**Comment by xorrvin** [comment]
Hey guys, just wanted to share yet another solution: https://github.com/xorrvin/gitoboros. Essentially it works the same way as other projects, but is available as a web version: https://gitoboros.xyz.

Cheers!

**Comment by SF-Prog** [comment]
I am having same issue, would be nice to have a neat solution for this scenarios into consideration, as many companies are currently leveraging security measures. We would like to show ourselves as active (contributions from work account into personal) but also to avoid the security team to point their fingers to us, or hold until we leave the company for this to happen.

**Reply by isturiz** [reply]
https://github.com/isturiz/sync-graph

Try that!

**Comment by isturiz** [comment]
I encountered this problem too, so I developed this solution: https://github.com/isturiz/sync-graph

It's simple to use with the CLI.

**Reply by angushhz** [reply]
It's help me resolve above problem. Anyone can try it !
https://github.com/isturiz/sync-graph/issues/2

**Comment by JelleKoster** [comment]
Using [these steps](https://saraford.net/2017/02/20/how-to-view-the-name-and-email-that-is-associated-with-a-commit-on-github-via-the-web-browser-051/) I found my old commits have a noreply e-mail address, probably because **at the time of the commits** my settings had the "keep my email addresses private" ticked.
Unticking this later does not reverse this for old commits.
As per the [documentation](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/merging-multiple-personal-accounts), commits using noreply e-mail addresses cannot be transferred.

Therefore the syncing "fake" re-commit solutions mentioned in the other is the only way to get the sync graph to show your level of activity visually, but it will not stand up to any inspection when people want to see those commits.

As an alternate solution, I have created an organisation with both my accounts as members.
That will allow me to always have a historical view back to the separate accounts I had and their individual commit activities.

**Comment by annettomcy** [comment]
Even though I followed all the steps, the contributions don't show. And also no, "keep my email addresses private" was not ticked on my work account. The contributions still show on the account made with work email.

**Comment by mrsarthakgupta** [comment]
GitHub doesn’t let you merge contribution graphs from two different accounts, and activity can’t be “copied” from one profile to another. The contribution graph is tied to the account and to the email that was used when commits were created.

If you want contributions from your work to appear on your personal profile, the only reliable way is:

→ Make sure your personal email is added as a verified email on the work account, or used in the commit author info.

GitHub counts a commit for every account that has that email verified.
So if the commits were made with an email you control (for example your personal one), the activity will show up on both accounts automatically.

If your work commits were made with your company email only, then:

You need to add that work email to your personal GitHub account and verify it before the company deletes it.

Once verified, all past commits associated with that email will also appear on your personal contribution graph.

If the company email gets deleted before you verify it, you won’t be able to recover those contributions — GitHub requires ownership of the email to link it.

**Reply by annettomcy** [reply]
So I added my work email, got it verified and also updated my commits to my personal email. I still see the contributions tied to the old account, even though both the emails are now tied to my personal acc.

**Reply by annettomcy** [reply]
Thanks for the response :")

**Reply by youennlj-sfl** [reply]
> GitHub counts a commit for every account that has that email verified.
So if the commits were made with an email you control (for example your personal one), the activity will show up on both accounts automatically.

You cannot add an e-mail address that is already tied to a GitHub account, so this advice does not work.

**Comment by mubtashimrafid** [comment]
GitHub contributions cannot be merged or transferred between two separate accounts. The contribution graph is tied to the account that made the commits, issues, pull requests, and other activity.

However, you may still be able to show some work contributions on your personal account by:

* Adding your work email address to your personal GitHub account as a verified secondary email
* Making sure future commits use an email connected to your personal account
* Using the same verified email for commits so GitHub can attribute them correctly
* Keeping repositories public (or enabling private contribution visibility in profile settings)

Past contributions that were already made under your work account generally cannot be moved to another account.

A good long-term approach is to use your personal account for contributions whenever company policy allows, or link to important work projects in your profile README or portfolio.

**Comment by Aryan-246** [comment]
Hello @catalincatrina,

Short answer: you **cannot transfer contribution history** (the green grid) from one account to another on GitHub. That’s not how their system is designed.

Now the important part most people miss:

Your contributions are **not actually tied to your account directly** — they’re tied to the **email address used in your commits**.

So here’s what actually works:

### What you should do

1. Make sure the **email used in your work commits** (your company email) is added to your personal account:

   * Go to Settings → Emails
   * Add your work email there

2. Even if that email gets deleted later, your past commits will still count — GitHub matches contributions based on commit metadata, not active email status.

3. (Optional but smart) Set your **personal account email as the primary commit email going forward**:

   ```bash
   git config --global user.email "your-personal-email@example.com"
   ```

### What will NOT work

* You cannot merge two GitHub accounts
* You cannot “move” contribution graphs
* You cannot retroactively reassign contributions across accounts unless you rewrite commit history (which is messy and risky for shared repos)

### Hard truth

If your work account used a different email and you **don’t add that email to your personal account**, then yes — your contribution graph on your personal profile will look empty.

But if you set it up correctly now, your work will still show on your personal account even after you lose access to the work account.

No hacks needed — just understand how GitHub tracks contributions.

Best regards.
